src/liballoc/collections/vec_deque.rs

```diff
@@ -1834,8 +1862,22 @@ impl<T> VecDeque<T> {
     #[inline]
     #[stable(feature = "append", since = "1.4.0")]
     pub fn append(&mut self, other: &mut Self) {
-        // naive impl
-        self.extend(other.drain(..));
+        unsafe {
+            // Guarantees there is space in `self` for `other`.
+            self.reserve(other.len());
+
+            {
+                let (src_high, src_low) = other.as_slices();
+
+                // This is only safe because copy_slice never panics when capacity is sufficient.
+                self.copy_slice(src_low);
+                self.copy_slice(src_high);
```

Comment:
Are these two lines in the wrong order?  `src_high` should be appended before `src_low`.